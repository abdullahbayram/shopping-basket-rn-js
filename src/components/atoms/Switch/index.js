import { Switch as RNSwitch } from 'react-native';

export default RNSwitch;
