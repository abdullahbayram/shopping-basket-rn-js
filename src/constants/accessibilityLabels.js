const accessibilityLabels = {
  search: 'Search Button',
  more: 'More Options Button',
  back: 'Back Button',
};

export default accessibilityLabels;
