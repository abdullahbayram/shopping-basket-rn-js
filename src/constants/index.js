export { default as strings } from './strings';
export { default as toastMessages } from './toastMessages';
export { default as urls } from './urls';
