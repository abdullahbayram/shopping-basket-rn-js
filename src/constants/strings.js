export default {
  icons: {
    cardholderName: 'account',
    cardholderNameCheck: 'account-check',
    expirationDate: 'calendar-alert',
    expirationDateCheck: 'calendar-check',
    cvv: 'shield-alert',
    cvvCheck: 'shield-check',
    creditCardNumber: 'credit-card',
    creditCardNumberCheck: 'credit-card-check',
  },
  screens: {
    success: 'Success',
    payment: 'Payment',
    error: 'Error',
    checkout: 'Checkout',
  },
  devErrors: {
    parseErrorMessage: 'Failed to parse error message:',
  },
  buttons: {
    retry: 'Retry',
    checkout: 'CHECKOUT',
    order: 'ORDER',
    payAndOrder: 'PAY AND ORDER',
    gotoProducts: 'Go to Products',
  },
  productList: {
    basketItemCount: 'Items in the basket: ',
    loading: 'Loading products...',
    errorLoading: 'Unable to load products. Please try again.',
    limitReached: 'You reached the max quantity per product!',
  },
  checkout: {
    total: 'Total:',
    order: 'ORDER',
    promoCodePlaceholder: 'Enter your promo code',
    promoCode: 'Promo Code',
    promoCodeRequired: 'Promo code can not be empty',
    promoCodeNotValid: 'This promo code is not valid',
    applyPromo: 'APPLY PROMO CODE',
    emptyBasket: 'Your basket is empty.',
  },
  payment: {
    total: 'Total:',
    basketItemCount: 'Items in the basket: ',
    cardholderName: 'Cardholder Name',
    cardholderNamePlaceholder: 'Enter the name on your card',
    creditCardNumber: 'Credit Card Number',
    creditCardPlaceholder: 'Enter your credit card number',
    expirationDate: 'Expiration Date',
    expirationDatePlaceholder: 'MM/YY',
    cvv: 'CVV',
    cvvPlaceholder: '123',
    unexpectedError: 'An unexpected error occurred. Please try again later.',
    unexpectedErrorShort: 'Unexpected Error',
    invalidCard: 'Invalid credit card number',
    cardHolderRequired: 'Cardholder name is required',
    creditCardRequired: 'Credit card number is required',
    expirationDateRequired: 'Expiration date is required',
    cvvRequired: 'CVV is required',
    invalidExpirationDate: 'Invalid expiration date (MM/YY)',
    cvvLength: 'CVV must be 3 digits',
    cardHolderMinLength: 'Cardholder name must be at least 3 characters',
    success: `
Thank you!
Your order has been placed successfully.
      `,
  },
};
