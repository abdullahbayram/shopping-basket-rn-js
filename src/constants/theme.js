import { MD3LightTheme, MD3DarkTheme } from 'react-native-paper';

export const sharedColors = {
  overlayBackgroundColor: 'rgba(0,0,0,0.5)',
};

const lightColors = {
  primary: '#3498db',
  secondary: '#FED034',
  tertiary: '#34300F',
  error: '#FF6B6B',
  background: '#FFF',
  surface: '#F9F9F9',
  textPrimary: '#004f6b',
  textSecondary: '#666',
  textTertiary: '#000',
  linearLeft: '#6fc8dc',
  linearRight: '#8bd4c7',
  containerBackground: '#F9F9F9',
  shadowColor: 'rgba(0, 0, 0, 0.3)',
  buttonBackground: '#FFF',
  buttonBorder: '#FFA500',
  transparentButtonBorder: '#FFA500',
  cardBackground: '#F0F0F0',
  ratingStar: '#FFA500',
  textFour: '#333',
  disabledButton: '#ddd',
  disabledButtonBorder: '#aaa',
  get spinner() {
    return this.secondary;
  },
  progressBarBackground: '#e0e0e0',
  progressBar: '#76c7c0',
};

const darkColors = {
  primary: '#2980b9',
  secondary: '#E0A800',
  tertiary: '#FFF',
  error: '#FF6B6B',
  background: '#121212',
  surface: '#1E1E1E',
  textPrimary: '#E0E0E0',
  textSecondary: '#B3B3B3',
  textTertiary: '#FFF',
  linearLeft: '#2980b9',
  linearRight: '#4fb0c7',
  containerBackground: '#333',
  shadowColor: 'rgba(255, 255, 255, 0.1)',
  buttonBackground: '#444',
  buttonBorder: '#FFAA33',
  transparentButtonBorder: '#FFAA33',
  cardBackground: '#424242',
  ratingStar: '#FFD700',
  textFour: '#FFF',
  disabledButton: '#555',
  disabledButtonBorder: '#777',
  get spinner() {
    return this.secondary;
  },
  progressBarBackground: '#444',
  progressBar: '#76c7c0',
};

export const lightTheme = {
  ...MD3LightTheme,
  roundness: 2,
  colors: {
    ...MD3LightTheme.colors,
    ...lightColors,
  },
};

export const darkTheme = {
  ...MD3DarkTheme,
  roundness: 2,
  colors: {
    ...MD3DarkTheme.colors,
    ...darkColors,
  },
};

export const spacing = {
  xxs: 2,
  xs: 4,
  sm: 10,
  md: 16,
  lg: 24,
  xl: 32,
  special: 7,
  baseScreenSpecial: 9,
};
