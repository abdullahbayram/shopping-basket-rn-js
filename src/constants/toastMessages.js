const messages = {
  basket: {
    limitReached: { title: 'Limit Reached', msg: 'You can only add a maximum of 15 units per item.' },
    empty: { title: 'Empty Basket', msg: 'Please add items to your basket before placing an order.' },
    invalidQuantity: { title: 'Invalid Quantity', msg: 'Quantity must be between 1 and 5.' },
  },
  promo: {
    success: { title: 'Promo Code', msg: 'Promo code applied successfully' },
    invalid: { title: 'Promo Code', msg: 'Invalid promo code' },
    error: { title: 'Promo Error', msg: 'Error applying promo code' },
  },
};
export default messages;
