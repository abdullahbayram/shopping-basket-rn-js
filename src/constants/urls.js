export const baseUrl = 'http://localhost:9001/';

export const products = 'products';
export const checkout = 'checkout';
