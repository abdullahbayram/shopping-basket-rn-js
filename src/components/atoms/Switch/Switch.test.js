import React from 'react';
import { render } from '@testing-library/react-native';
import Switch from '.';
import { lightColors } from '../../../constants/theme';

test('Switch renders correctly', () => {
  const view = render(
    <Switch
      trackColor={{ false: lightColors.textPrimary, true: lightColors.textSecondary }}
      ios_backgroundColor={lightColors.textPrimary}
      value
    />,
  );
  expect(view.toJSON()).toMatchSnapshot();
});
