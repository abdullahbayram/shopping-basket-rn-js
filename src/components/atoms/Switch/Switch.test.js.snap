// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders correctly 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#666"
  style={
    [
      {
        "height": 31,
        "width": 51,
      },
      {
        "backgroundColor": "#004f6b",
        "borderRadius": 16,
      },
    ]
  }
  tintColor="#004f6b"
  value={true}
/>
`;
